src/main/java/io/github/genomicdatainfrastructure/daam/mappers/RemsApplicationMapper.java

```diff
@@ -6,157 +6,241 @@
 
 import io.github.genomicdatainfrastructure.daam.model.*;
 import io.github.genomicdatainfrastructure.daam.remote.rems.model.*;
+import jakarta.enterprise.context.ApplicationScoped;
 
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
 import java.util.Set;
 
-public class RemsApplicationMapper {
+@ApplicationScoped
+public class RemsApplicationMapper implements ApplicationMapper<Application> {
```

Comment:
important: this `RemsApplicationMapper` is a concrete class highly coupled with REMS api implementation. So it should much closer to it.

I liked the idea, but perhaps this interface should be about integration with REMS directly and not just about parsing POJOs / DTOs. I've got an idea how to fix it with clean architecture, but I would like to validate it with you.